core/src/main/java/de/evoila/cf/broker/interceptor/OriginatingIdentityInterceptor.java

```diff
@@ -0,0 +1,50 @@
+package de.evoila.cf.broker.interceptor;
+
+import de.evoila.cf.broker.model.ApiVersions;
+import de.evoila.cf.broker.model.annotations.ApiVersion;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.method.HandlerMethod;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.resource.ResourceHttpRequestHandler;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class OriginatingIdentityInterceptor implements HandlerInterceptor {
+
+    private static final String X_BROKER_API_REQUEST_IDENTITY = "X-Broker-API-Request-Identity";
+    private final Logger log = LoggerFactory.getLogger(OriginatingIdentityInterceptor.class);
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+
+        if (!(handler instanceof ResourceHttpRequestHandler)) {
+            HandlerMethod method = (HandlerMethod) handler;
+            String[] api;
+            log.info("Intercepting on method " + method.getMethod().getName());
+
+            if (method.hasMethodAnnotation(ApiVersion.class)) {
```

Comment:
Why only work with classes that have an API handler annotation? Does this header rely on that?